### PR TITLE
Add 'Client Migration to Keycloak' section

### PR DESCRIPTION
**Pull request by nikosev**
# Summary

This PR introduces the following changes:

- Add Migration guide to Keycloak and update the development environment endpoints for OIDC
- Add section for "Client Credentials"

### CODE REVIEW

File: content/en/providers/check-in/sp/_index.md [hunk 1]
```diff
@@ -866,6 +909,36 @@ $sessionName = "simple-oidc-client-php";
 $sessionLifetime = 60*60;  // must be equal to access token validation time in seconds
 ```
 
+### Client Migration to Keycloak
+
+Currently we have migrated only the Development instance of Check-in OP to
+Keycloak. After the successful migration of the Development instance of Check-in
+OP we will migrate the Demo and the Production instances.
```
Comment (NicolasLiampotis) [inline content/en/providers/check-in/sp/_index.md:916]: Replace with alert box containing the following text:

```
The migration guide below applies to OIDC clients registered in the **Development** 
environment of Check-in. **Beginning June XX, 2022 client using the legacy Check-in OIDC 
endpoints we will no longer be supported**
```

File: content/en/providers/check-in/sp/_index.md [hunk 2]
```diff
@@ -263,12 +263,20 @@ IdP provides an OpenID Connect (OAuth2) API based on
 [MITREid Connect](https://github.com/mitreid-connect/OpenID-Connect-Java-Spring-Server),
 which has been
 [certified by the OpenID Foundation](http://openid.net/certification/).
+Currently we are planning to migrate the EGI Check-in OIDC Provider from MITREid
+to [Keycloak](https://www.keycloak.org/). You can check the
+[Migration guide](#client-migration-to-keycloak) for more information.
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md:268]: done

File: content/en/providers/check-in/sp/_index.md [hunk 3]
```diff
@@ -263,12 +263,20 @@ IdP provides an OpenID Connect (OAuth2) API based on
...
 Interconnection with the EGI Check-in OIDC Provider allows users to sign in
 using any of the supported backend authentication mechanisms, such as
 institutional IdPs registered with eduGAIN or Social Providers. Once the user
 has signed in, EGI Check-in can return OIDC Claims containing information about
 the authenticated user.
 
+{{% alert title="Important" color="warning" %}} The EGI Check-in OIDC Provider
+has been migrated to Keycloak. Please check
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md:276]: done

File: content/en/providers/check-in/sp/_index.md [hunk 4]
```diff
@@ -362,15 +372,15 @@ The most important OIDC/OAuth2 endpoints are listed below:
...
+| Token                  | <https://aai.egi.eu/oidc/token>                            | <https://aai-demo.egi.eu/oidc/token>                            | <https://aai-dev.egi.eu/auth/realms/egi/protocol/openid-connect/token>            |
+| Device Code            | <https://aai.egi.eu/oidc/devicecode>                       | <https://aai-demo.egi.eu/oidc/devicecode>                       | <https://aai-dev.egi.eu/auth/realms/egi/protocol/openid-connect/auth/device>      |
+| JSON Web Key(JWK)      | <https://aai.egi.eu/oidc/jwk>                              | <https://aai-demo.egi.eu/oidc/jwk>                              | <https://aai-dev.egi.eu/auth/realms/egi/protocol/openid-connect/certs>            |
+| User Info              | <https://aai.egi.eu/oidc/userinfo>                         | <https://aai-demo.egi.eu/oidc/userinfo>                         | <https://aai-dev.egi.eu/auth/realms/egi/protocol/openid-connect/userinfo>         |
+| Introspection          | <https://aai.egi.eu/oidc/introspect>                       | <https://aai-demo.egi.eu/oidc/introspect>                       | <https://aai-dev.egi.eu/auth/realms/egi/protocol/openid-connect/token/introspect> |
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md:383]: done

File: content/en/providers/check-in/sp/_index.md [hunk 5]
```diff
@@ -866,6 +909,36 @@ $sessionName = "simple-oidc-client-php";
 $sessionLifetime = 60*60;  // must be equal to access token validation time in seconds
 ```
 
+### Client Migration to Keycloak
+
+Currently we have migrated only the Development instance of Check-in OP to
+Keycloak. After the successful migration of the Development instance of Check-in
+OP we will migrate the Demo and the Production instances.
+
+#### How to Migrate your Service to Keycloak
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md:918]: done

File: content/en/providers/check-in/sp/_index.md [hunk 6]
```diff
@@ -569,6 +579,40 @@ curl -X POST "${TOKEN_ENDPOINT}" \
 {{% alert title="Note" color="info" %}} You can find the _Token Endpoint_ in the
 [Endpoints](#endpoints) table.{{% /alert %}}
 
+##### Client Credentials
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md:582]: done

File: content/en/providers/check-in/sp/_index.md [hunk 7]
```diff
@@ -866,6 +872,75 @@ $sessionName = "simple-oidc-client-php";
 $sessionLifetime = 60*60;  // must be equal to access token validation time in seconds
 ```
 
+### Client Migration to Keycloak
+
+The migration guide below applies to OIDC clients registered in the
+**Development** environment of Check-in. **Beginning June XX, 2022 client using
```
Comment (NicolasLiampotis) [inline content/en/providers/check-in/sp/_index.md:878]: XX => 10
client => clients

File: content/en/providers/check-in/sp/_index.md [hunk 8]
```diff
@@ -866,6 +872,75 @@ $sessionName = "simple-oidc-client-php";
...
+the legacy Check-in OIDC endpoints we will no longer be supported**
```
Comment (NicolasLiampotis) [inline content/en/providers/check-in/sp/_index.md:879]: we will no longer => will no longer be

**Pull request closed without merging**